### DISCUSSION TITLE
NoeMCU LoLin3 doesn't boot with DS18B20 connected

### DISCUSSION THREAD

**Discussion by ghost** [arendst/Tasmota#13027] [Q&A] [answered]
I have this odd problem, need help.

Tasmota 9.5 on NodeMCU LoLin version3 board, connected with DS18B20 air temperature 3-wire sensor on GPIO2.

When cold start (with power off and on) with sensor connected, it will not boot up. If do so while sensor is disconnected, it will boot up OK. After cold start (with disconnected sensor), reconnecting sensor will not auto detect it. A soft restart is needed.

If soft restart (restart from Web UI), it will boot up OK, even with sensor connected.

So now each time there is a power cycle, I need to disconnect the sensor, power up, reconnected sensor, login to web UI, click restart button.

Why is this happening?

**Comment by barbudor** [comment, marked as answer]
GPIO2 is a special pin regarding boot and musn't tied down during boot.
I haven't seen this behavior before so I would suspect a cabling problem. 
At least try with another pin.

**Reply by ghost** [reply]
I've tried a few other pins, but none detected the sensor except GPIO2 (directly connected) and GPIO5 (only can work with resistor). they still need soft reboot to detect. I've also tried with another sensor (I have a few DS18B20), same result.

I'm at loss.

**Reply by sfromis** [reply]
To use DS18B20, you need a pullup resistor. With one DS18B20 and a not too long wire, the weak internal pullup usually does the job fine, you can activate that using `SetOption74 1`. Due to the requirement not to be pulled down at boot, gpio2 will normally have an external pullup resistor, which "helps" DS18B20.

**Reply by ghost** [reply]
I'm using `GPIO5 `now, without resistor after `SetOption74 1`. The problem still persists, unable to boot up with sensor connected. Only works when reconnected after boot up, and soft reboot.

**Reply by barbudor** [reply]
So you have another problem

**Reply by ghost** [reply]
Indeed it's another problem. The connector was the problem. I replaced the connector, and it works now. Thanks for the SetOption74, I can now use DS18B20 without pullup resistor.
